### PR TITLE
Add dialogs for twitter sync

### PR DESCRIPTION
**Pull request by DerekBurch**
Tasks
https://app.asana.com/0/20751731968782/136829453549310
https://app.asana.com/0/20751731968782/136829453549308

@andrewconner @cvializ

### CODE REVIEW

**Comment by andrewconner** [comment]
Nice 💣. Before you merge, let me know and I can make sure users don't have those prefs set

File: kifi-backend/modules/shoebox/angular/src/common/services/extensionLiaison.js [hunk 1]
```diff
@@ -13,9 +13,22 @@ angular.module('kifi')
           case 'get_guide':
             triggerGuide();
             break;
+          case 'guide_end':
+            $rootScope.$emit('guideEnd');
+            break;
           case 'import_bookmarks':
             if (data.count > 0) {
               $rootScope.$emit('showGlobalModal', 'importBookmarks', {msgEvent: event});
+              var deregisterModalClosed = $rootScope.$on('modalClosed', function () {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/extensionLiaison.js:22]: Oh fun, cool way to do it.

**Comment by andrewconner** [comment]
Looks good to me. @cvializ your vote of confidence?

**Comment by cvializ** [comment]
☑️ <- looks good!

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 2]
```diff
@@ -275,18 +275,28 @@ angular.module('kifi')
...
+    $scope.$on('$destroy', deregisterGuideEnd);
+
     function tryToTriggerTwitterPromo() {
       var guide = angular.element('.kifi-guide-4-stage, .kifi-guide-0-stage').length;
       if (!guide) {
-        // do it.
-        // Consider immediately setting twitter_sync_promo to null.
-        angular.noop();
+        // Disabling dialog one until we are ready
+        if (false && profileService.prefs.twitter_sync_promo === 'show_sync') {
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: `false && ...`, dev condition?

Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: ah, I see the comment

**Comment by andrewconner** [comment]
I reset the flags for all users, @DerekBurch, so feel free to push this live. I have a companion PR #21428

**Comment by andrewconner** [comment]
(So no `false &&` needed if mine is merged with it)

**Pull request merged**